src/components/SignupForm/SignupForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { Button } from '@material-ui/core';
+import SignUpFormDataInterface from './interfaces/SignupFormDataInterface';
+import ErrorMessageVisibleInterface from './interfaces/ErrorMessageVisibleInterface';
+import useSignupFormValidation from './useSignupFormValidation';
+import useFormFieldsConfig from './useFormFieldsConfig';
+import FormField from './FormField';
+import { auth } from '../../firebase';
+
+const SignupForm: React.FC = () => {
+  const initialFormData: SignUpFormDataInterface = {
+    name: '',
+    password: '',
+    repeatPassword: '',
+    email: ''
+  };
+
+  const initialErrorMessageVisible: ErrorMessageVisibleInterface = {
+    name: false,
+    email: false,
+    password: false,
+    repeatPassword: false
+  };
+
+  const [formData, setFormData] = useState(initialFormData);
+  const [errorMessageVisible, setErrorMessageVisible] = useState(initialErrorMessageVisible);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { error, setError, validateForm } = useSignupFormValidation(formData);
+  const formFieldsConfig = useFormFieldsConfig(formData, errorMessageVisible);
```

Comment:
You should use useMemo here because we don't want to call this function again for example when changed is `isLoading`